src/parser.cc

```diff
@@ -1570,6 +1570,25 @@ bool readXml(tinyxml2::XMLElement *_xml, ElementPtr _sdf,
     for (elemXml = _xml->FirstChildElement(); elemXml;
          elemXml = elemXml->NextSiblingElement())
     {
+      // Expand URIs via user-provided callback if specified by configuration
+      if (std::string("uri") == elemXml->Value() && _config.StoreResolvedURIs())
+      {
+        const std::string uri = elemXml->GetText();
+        const std::string resolvedUri = sdf::findFile(uri, true, true, _config);
+
+        if (resolvedUri.empty())
+        {
+          Error err(ErrorCode::URI_LOOKUP, "Unable to resolve uri[" + uri + "]",
+              _source, elemXml->GetLineNum());
+          err.SetXmlPath(_sdf->XmlPath() + "/uri");
+          _errors.push_back(err);
+        }
+        else
+        {
+          elemXml->SetText(resolvedUri.c_str());
```

Comment:
This means we'll lose the original URI, which is not ideal for certain use cases, e.g. saving a world to sdf. A few alternatives to consider:
1. Do the file resolution in `sdf::Mesh`. This requires passing the `ParserConfig` down to `sdf::Mesh` from `sdf::Model`. Tedious, but doable, and wouldn't break ABI.
2. Add a member variable to `sdf::ElementPrivate` to store the resolved URI and expose this variable through `sdf::Mesh`,  Unfortunately, this would break ABI since `ElementPrivate` is used by templates such as `sdf::Element::Get`.
3. (Hacky) Add the resolved URI as a custom attribute to the `<uri>` element, e.g. `gz:resolved_uri="path/to/file"`. Then read this attribute in `sdf::Mesh`. Note that this will likely be included in the generated SDF when saving worlds, but that won't break anything since it's a custom attribute.

Reply:
I ended up going with approach 1.  In addition to mesh, skybox and heightmap also make use of URI elements, so those were resolved as well.